### PR TITLE
Dedupe entries when inserting

### PR DESCRIPTION
**Pull request by nyarly**

### CODE REVIEW

**Comment by sous-machine-account** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR: 626 lines of code changed</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: util/sqlgen/field.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+		if !other[n].equal(v) {
+			return false
+		}
+	}
+	for n, v := range other {
+		if !r[n].equal(v) {
+			return false
+		}
+	}
+	return true
```
Comment (lachlanh) [inline util/sqlgen/field.go]: does this account for row and other having different lengths ? could read over the end of one ? also not sure why we test both sides ?

**Comment by lachlanh** [comment]
made a comment on the equals method for rows, rest of the code mainly looks like refactoring to the sqlgen package.

File: util/sqlgen/field.go (hunk 1 above)
Comment (nyarly) [reply util/sqlgen/field.go]: They're maps - you can't len(map). We check both ways in order to be sure that we're checking the union of both of their keys.

File: util/sqlgen/fieldset_test.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package sqlgen
+
+import "testing"
+
+func TestRowDedupe(t *testing.T) {
```
Comment (nyarly) [inline util/sqlgen/fieldset_test.go]: @lachlanh This is the motivation for this PR.

**Pull request merged**